wordpress-mobile/WordPress-Android
# Unified Login & Sign-Up: Prologue screen UI update

Fixes #11782

Depends on #11780 

## Screens

### Phone

| Portrait | Landscape | 
|---|---|
|![phone-api27-light-portrait](https://user-images.githubusercontent.com/830056/81459263-66caf680-9175-11ea-9e36-33bce0c96964.png)|![phone-api27-light-landscape](https://user-images.githubusercontent.com/830056/81459261-629ed900-9175-11ea-8c13-b60b90973c06.png)|

### Tablet

| Portrait | Landscape | 
|---|---|
|![tablet-api27-light-portrait](https://user-images.githubusercontent.com/830056/81459360-e35dd500-9175-11ea-9645-4a453fb3f210.png)|![tablet-api27-light-landscape](https://user-images.githubusercontent.com/830056/81459354-de008a80-9175-11ea-84be-b33607da4035.png)|

### Dark Theme

| Portrait | Landscape | 
|---|---|
|![phone-api27-dark-portrait](https://user-images.githubusercontent.com/830056/81459393-11dbb000-9176-11ea-8756-9e5f005137c0.png)|![phone-api27-dark-landscape](https://user-images.githubusercontent.com/830056/81459388-0c7e6580-9176-11ea-9ff3-e21945a54854.png)|

### Status and Navigation Bars Variations

#### Light Theme

| API 21+ | API 23+ | API 27+ |
|---|---|---|
|![phone-api21-light-portrait](https://user-images.githubusercontent.com/830056/81459430-4485a880-9176-11ea-8e3b-15942e69b678.png)|![phone-api23-light-portrait](https://user-images.githubusercontent.com/830056/81459435-49e2f300-9176-11ea-9a12-a60c4f36ac71.png)|![phone-api27-light-portrait](https://user-images.githubusercontent.com/830056/81459263-66caf680-9175-11ea-9e36-33bce0c96964.png)

#### Dark Theme

| API 21+ | API 23+ | API 27+ |
|---|---|---|
|![phone-api21-dark-portrait](https://user-images.githubusercontent.com/830056/81459469-81ea3600-9176-11ea-8304-f437ced32e87.png)|![phone-api23-dark-portrait](https://user-images.githubusercontent.com/830056/81459500-8adb0780-9176-11ea-96c4-b5470aadda18.png)|![phone-api27-dark-portrait](https://user-images.githubusercontent.com/830056/81459393-11dbb000-9176-11ea-8756-9e5f005137c0.png)|

## To test

Simply start the app and make sure the Prologue screen layout looks like the ones provided by the design spec (internal ref: pbArwn-mA-p2). 

Repeat that for different phone and tablet sizes, in both portrait and landscape orientations, and experiment with the Light and Dark themes on different API levels.

## Notes
- I decided not to use any feature flags here, as it didn't seem likely that these changes could introduce some kind of critical bug – in fact, maintaining two different sets of layouts/styles/themes for the same screen could even increase that risk – and even if that happens, it seems that it would be easier to just revert the changes.
- That being said, setting the `UNIFIED_LOGIN_AVAILABLE` flag to `false` should still make the old flow available, but with the new UI.
- A lot of changes here were caused by the removal of the old ViewPager pages used by the Prologue screen. Their strings, animation files, tracking events and much of the adapter logic have been removed and will be replaced in the near future.
- The ViewPager itself is not being removed, but its page indicator is being hidden for now. Despite that, the indicator size is still taken into account when measuring the layout, which is why the bottom margin of the illustration might look weird on smaller screens or in landscape mode.
- The Lottie library dependency is also not being removed for now, as we're still not sure if we're going to use animations in our new pages. That being said, removing it now is still a fair option.

---

PR submission checklist:
